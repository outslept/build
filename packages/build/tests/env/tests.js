import { platform } from 'process'

import { Fixture, normalizeOutput } from '@netlify/testing'
import { test, expect } from 'vitest'

test('Environment variables are set in plugins', async () => {
  const output = await new Fixture('./fixtures/plugin').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

// Windows environment variables work differently
if (platform !== 'win32') {
  test('Environment variable in build.command', async () => {
    const output = await new Fixture('./fixtures/command').runWithBuild()
    expect(normalizeOutput(output)).toMatchSnapshot()
  })
}

test('build.environment are set in plugins', async () => {
  const output = await new Fixture('./fixtures/build').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('Does not pass BUGSNAG_KEY to build command and plugins', async () => {
  const output = await new Fixture('./fixtures/bugsnag_key')
    .withEnv({ BUGSNAG_KEY: '00000000000000000000000000000000' })
    .runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('BRANCH environment variables is set with --branch', async () => {
  const output = await new Fixture('./fixtures/branch').withFlags({ branch: 'test' }).runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('DEPLOY_ID environment variables is set with --deploy-id', async () => {
  const output = await new Fixture('./fixtures/deploy_id').withFlags({ deployId: 'test' }).runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('BUILD_ID environment variables is set with --build-id', async () => {
  const output = await new Fixture('./fixtures/build_id').withFlags({ buildId: 'test-build' }).runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('CONTEXT environment variables is set with --context', async () => {
  const output = await new Fixture('./fixtures/context').withFlags({ context: 'test' }).runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})
