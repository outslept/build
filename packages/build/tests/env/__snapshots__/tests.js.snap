// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BRANCH environment variables is set with --branch 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  branch: test
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/branch
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/branch

> Config file
  packages/build/tests/env/fixtures/branch/netlify.toml

> Resolved config
  build:
    environment:
      - TEST
    publish: packages/build/tests/env/fixtures/branch
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  production

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

test

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`BUILD_ID environment variables is set with --build-id 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  buildId: test-build
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/build_id
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/build_id

> Config file
  packages/build/tests/env/fixtures/build_id/netlify.toml

> Resolved config
  build:
    environment:
      - TEST
    publish: packages/build/tests/env/fixtures/build_id
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  production

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

test-build

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`CONTEXT environment variables is set with --context 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  context: test
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/context
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/context

> Config file
  packages/build/tests/env/fixtures/context/netlify.toml

> Resolved config
  build:
    environment:
      - TEST
    publish: packages/build/tests/env/fixtures/context
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  test

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

test

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`DEPLOY_ID environment variables is set with --deploy-id 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  deployId: test
  repositoryRoot: packages/build/tests/env/fixtures/deploy_id
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/deploy_id

> Config file
  packages/build/tests/env/fixtures/deploy_id/netlify.toml

> Resolved config
  build:
    environment:
      - TEST
    publish: packages/build/tests/env/fixtures/deploy_id
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  production

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

test

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`Does not pass BUGSNAG_KEY to build command and plugins 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/bugsnag_key
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/bugsnag_key

> Config file
  packages/build/tests/env/fixtures/bugsnag_key/netlify.toml

> Resolved config
  build:
    command: node index.js
    commandOrigin: config
    publish: packages/build/tests/env/fixtures/bugsnag_key
    publishOrigin: default

> Context
  production

build.command from netlify.toml                               
────────────────────────────────────────────────────────────────

$ node index.js
undefined

(build.command completed in 1ms)
Build step duration: build.command completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`Environment variables are set in plugins 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/plugin
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/plugin

> Config file
  packages/build/tests/env/fixtures/plugin/netlify.toml

> Resolved config
  build:
    publish: packages/build/tests/env/fixtures/plugin
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  production

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

en_US.UTF-8

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`build.environment are set in plugins 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/env/fixtures/build
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/env/fixtures/build

> Config file
  packages/build/tests/env/fixtures/build/netlify.toml

> Resolved config
  build:
    environment:
      - TEST
    publish: packages/build/tests/env/fixtures/build
    publishOrigin: default
  plugins:
    - inputs: {}
      origin: config
      package: ./plugin

> Context
  production

> Loading plugins
   - ./plugin@1.0.0 from netlify.toml

./plugin (onPreBuild event)                                   
────────────────────────────────────────────────────────────────

test

(./plugin onPreBuild completed in 1ms)
Build step duration: ./plugin onPreBuild completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;
